pg_analytics/src/hooks/createfunction.rs

```diff
@@ -0,0 +1,220 @@
+            for row_index in 0..num_rows {
+                for arg_index in 1..num_args {
+                    let fcinfo_arg = (*fcinfo).args.as_mut_ptr().add(arg_index - 1);
+                    (*fcinfo_arg).value = arg_arrays[arg_index - 1]
+                        .get_datum(row_index)?
+                        .ok_or(DataFusionError::Internal("No datum for type".to_string()))?;
+                    (*fcinfo_arg).isnull = false;
+                }
+
+                let result_datum = (*(*fcinfo).flinfo)
+                    .fn_addr
+                    .ok_or(DataFusionError::Internal(
+                        "Invalid function address for udf".to_string(),
+                    ))?(fcinfo);
+                result_vec.push((result_datum, false));
+            }
+
+            // UDF return type
+            let rettype = pg_sys::get_func_rettype(func_oid);
+
+            Ok(ColumnarValue::Array(
+                result_vec
+                    .into_iter()
+                    .into_arrow_array(rettype.into(), PgTypeMod(-1))?,
+            ))
+        } else {
+            Err(DataFusionError::Internal("No funcname".to_string()))
+        }
+    });
+
+    drop(memory_context);
+
+    ret
+}
+
+pub unsafe fn loadfunction(funcname: &str) -> Result<(), ParadeError> {
+    // Register all overloads with this function name
+    let mut func_candidate = func_list_from_name(funcname)?;
+    while !func_candidate.is_null() {
+        let func_oid = (*func_candidate).oid;
+        let arg_types = (*func_candidate).args.as_mut_ptr();
+        let nargs = (*func_candidate).nargs;
+        let ret_oid = pg_sys::get_func_rettype(func_oid);
+
+        // Create vector of input types
+        let mut input_types = vec![];
+        input_types.push(DataType::Utf8); // function name
+        for param_index in 0..nargs {
+            let arg_oid = arg_types.add(param_index as usize);
+            let ArrowDataType(datatype) =
+                PgAttribute((*arg_oid).into(), PgTypeMod(-1)).try_into()?;
+            input_types.push(datatype);
+        }
+
+        let ArrowDataType(return_type) = PgAttribute(ret_oid.into(), PgTypeMod(-1)).try_into()?;
+
+        let udf = create_udf(
+            funcname,
+            input_types,
+            Arc::new(return_type),
+            Volatility::Immutable,
+            Arc::new(|args| unsafe { udf_datafusion(args) }),
+        );
+
+        Session::with_session_context(|context| {
+            Box::pin(async move {
+                context.register_udf(udf);
+                Ok(())
+            })
+        })?;
+
+        func_candidate = (*func_candidate).next;
+    }
+
+    Ok(())
+}
+
+pub unsafe fn createfunction(
+    createfunction_stmt: *mut pg_sys::CreateFunctionStmt,
+) -> Result<(), ParadeError> {
+    // Drop any functions with the same name from the context and allow the next call to the UDF load them back
+
+    let funcname = pg_sys::NameListToString((*createfunction_stmt).funcname);
```

Comment:
If this is a no-op, let's just return `Ok()`

Reply:
Normally I would agree, but I wanted to leave this in the format that we would need it in when it's fully implemented.